index.d.ts

```diff
@@ -1,17 +1,20 @@
-import { Dispatch, Store, Unsubscribe } from 'redux';
+import { Dispatch, Store, Unsubscribe, AnyAction } from 'redux';
 
 export function shallowEquals(a: any, b: any): boolean;
 
-interface Options {
+type Options = {
   skipInitialCall?: boolean;
   equals?: typeof shallowEquals;
 }
 
-type Observer<S = any, MS = any> = unknown;
-type Mapper<S = any, MS = any> = (state: S) => MS;
-type Dispatcher<MS> = (dispatch: Dispatch<MS>, currentState: MS, previousState: MS) => unknown;
+type Mapper<S, MS> = (state: S) => MS;
+type Observer<S = any> = (state: S, dispatch: Dispatch<AnyAction>) => void;
+type Dispatcher<MS> = (dispatch: Dispatch<AnyAction>, current: MS, previous?: MS) => void;
 
-export function observer<S, MS>(mapper: Mapper<S, MS>, dispatcher: Dispatcher<MS>, options?: Options): Observer<S, MS>;
-export function observer<S>(dispatcher: Dispatcher<S>, options?: Options): Observer<S, S>;
+interface ObserverCreator {
+    <S>(dispatcher: Dispatcher<S>, locals?: Options): Observer<S>
```

Comment:
`Dispatcher` is defined as `Dispatcher<MS>` above, but here we're using `<S>` instead, which I would assume to be the `store` state and not a mapped state (which is what the dispatcher is called with at runtime). Is this important or am I reading too much into it?